File: apps/api/internal/models/site_settings.go
package models

import (
	"time"

	"gorm.io/datatypes"
	"gorm.io/gorm"

	"megagig-software-solution/apps/api/internal/ids"
)

// SiteSettings is a singleton — exactly one row ever exists. Contact
// details, hero copy, and pricing blurbs live here so they can change
// without a redeploy (see architecture.md rule #5).
type SiteSettings struct {
	ID             string         `gorm:"primarykey;size:36" json:"id"`
	ContactEmail   string         `gorm:"size:255" json:"contact_email"`
	ContactPhone   string         `gorm:"size:50" json:"contact_phone"`
	WhatsAppNumber string         `gorm:"size:50" json:"whatsapp_number"`
	Address        string         `gorm:"size:500" json:"address"`
	SocialLinks    datatypes.JSON `gorm:"type:jsonb" json:"social_links"`
	HeroHeadline   string         `gorm:"size:255" json:"hero_headline"`
	HeroSubhead    string         `gorm:"size:500" json:"hero_subhead"`
	PricingBlurbs  datatypes.JSON `gorm:"type:jsonb" json:"pricing_blurbs"`

	// About & founder copy — read by Home's OurStory/FounderSpotlight and
	// the /about-us page. FounderBio paragraphs are separated by a blank
	// line; empty fields simply hide their section on the public site.
	MissionStatement   string `gorm:"size:500" json:"mission_statement"`
	FoundedYear        int    `json:"founded_year"`
	FoundingStory      string `gorm:"type:text" json:"founding_story"`
	FounderName        string `gorm:"size:255" json:"founder_name"`
	FounderRole        string `gorm:"size:255" json:"founder_role"`
	FounderQuote       string `gorm:"size:500" json:"founder_quote"`
	FounderBio         string `gorm:"type:text" json:"founder_bio"`
	FounderPhotoURL    string `gorm:"size:500" json:"founder_photo_url"`
	FounderGithubURL   string `gorm:"size:500" json:"founder_github_url"`
	FounderLinkedinURL string `gorm:"size:500" json:"founder_linkedin_url"`
	FounderTwitterURL  string `gorm:"size:500" json:"founder_twitter_url"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// BeforeCreate generates a UUID before inserting.
func (s *SiteSettings) BeforeCreate(tx *gorm.DB) error {
	if s.ID == "" {
		s.ID = ids.New()
	}
	return nil
}

File: packages/shared/types/site-settings.ts
export interface SocialLinks {
  linkedin?: string;
  github?: string;
  youtube?: string;
  facebook?: string;
  twitter?: string;
}

export interface SiteSettings {
  id: string;
  contact_email: string;
  contact_phone: string;
  whatsapp_number: string;
  address: string;
  social_links: SocialLinks | null;
  hero_headline: string;
  hero_subhead: string;
  pricing_blurbs: Record<string, string> | null;
  // About & founder copy. Empty string = not set (the public site hides
  // the matching section). founder_bio paragraphs are separated by a blank line.
  mission_statement: string;
  founded_year: number;
  founding_story: string;
  founder_name: string;
  founder_role: string;
  founder_quote: string;
  founder_bio: string;
  founder_photo_url: string;
  founder_github_url: string;
  founder_linkedin_url: string;
  founder_twitter_url: string;
  created_at: string;
  updated_at: string;
}
